trace-sdk/src/androidTest/java/io/bitrise/trace/data/collector/device/DeviceCarrierDataCollectorInstrumentedTest.java

```diff
@@ -2,26 +2,28 @@
 
 import org.junit.Test;
 
-import io.bitrise.trace.data.dto.Data;
 import io.bitrise.trace.data.collector.BaseDataCollectorInstrumentedTest;
+import io.bitrise.trace.data.dto.Data;
 
+import static org.hamcrest.CoreMatchers.equalTo;
 import static org.hamcrest.CoreMatchers.is;
-import static org.hamcrest.CoreMatchers.notNullValue;
 import static org.hamcrest.MatcherAssert.assertThat;
 
 /**
  * Instrumented tests for {@link DeviceCarrierDataCollector}.
  */
 public class DeviceCarrierDataCollectorInstrumentedTest extends BaseDataCollectorInstrumentedTest {
 
+    final DeviceCarrierDataCollector collector = new DeviceCarrierDataCollector(context);
+
     /**
      * Verifies that when {@link DeviceCarrierDataCollector#collectData()} is called, the content of the returned
-     * {@link Data} should not be {@code null}.
+     * {@link Data} should match the expected data.
      */
     @Test
     public void collectData_contentShouldBeNotNull() {
-        final DeviceCarrierDataCollector deviceCarrierDataCollector = new DeviceCarrierDataCollector(context);
-        final String actualValue = (String) deviceCarrierDataCollector.collectData().getContent();
-        assertThat(actualValue, is(notNullValue()));
+        final Data expectedData = new Data(DeviceCarrierDataCollector.class);
+        expectedData.setContent("Android");
```

Comment:
Test will fail if tested on real devices, or if the given emulated device does not have a sim card. I think we should keep the test as is. A new test can be added, if we check that the device has a sim card and is an emulated device.